gridfs_bucket.go

```diff
@@ -105,26 +103,8 @@ func (b *gridfsBucket) Join(elems ...string) string { return consistentJoin(elem
 // supports a single read and a single write deadline for a bucket instance.
 // This function sets the read and write deadlines to allow it to respect the
 // context timeouts passed in by the caller.
-func (b *gridfsBucket) bucket(ctx context.Context) (*gridfs.Bucket, error) {
-	if err := ctx.Err(); err != nil {
-		return nil, errors.Wrap(err, "fetching bucket with canceled context")
-	}
-
-	gfs, err := gridfs.NewBucket(b.client.Database(b.opts.Database), options.GridFSBucket().SetName(b.opts.Name))
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
-
-	// The streaming GridFS functions do not accept a context so we need to
-	// set the deadline from the passed-in context here, if it exists, in
-	// order to respect context timeouts passed in by the caller.
-	dl, ok := ctx.Deadline()
-	if ok {
-		_ = gfs.SetReadDeadline(dl)
-		_ = gfs.SetWriteDeadline(dl)
-	}
-
-	return gfs, nil
+func (b *gridfsBucket) bucket() (*mongo.GridFSBucket, error) {
+	return b.client.Database(b.opts.Database).GridFSBucket(options.GridFSBucket().SetName(b.opts.Name)), nil
```

Comment:
The GridFSBucket now accepts a context, so we do not set the read and write deadline anymore

Reply:
Now accepts a context in it's async operations (seen below)